### PR TITLE
[OSSM-12223] Fix tmp invalid KialiCR after update MultiTenanty to ClusterWide in SMCP

### PR DESCRIPTION
**Pull request by mkralik3**
The memberroll controller sets `accessible_namespaces` and `api.namespaces.exclude` according to the SMCP mode here: https://github.com/maistra/istio-operator/blob/maistra-2.6/pkg/controller/servicemesh/memberroll/controller.go#L377-L381 

When a new SMCP is created, there is no issue:
- The Maistra operator generates the Kiali Custom Resource (CR) without `cluster_wide_access` set. (since it is not in kiali cr helm template)
- The memberroll controller populates `accessible_namespaces`.
- The `addons.go` logic then sets the `cluster_wide_access` field explicitly. This sequence does not cause issues.

However, a temporary invalid Kiali CR state can occur when an existing MultiTenant SMCP is updated to ClusterWide mode. In this scenario:
- The Kiali CR already exists with `cluster_wide_access=false`.
- When the memberroll controller updates `accessible_namespaces` to `**`, the Kiali CR becomes temporarily invalid from the Kiali operator's perspective. ( because contains accessible_namespaces=** as well as cluster_wide_access=false )
- This invalid state persists until the `addons.go` logic patches the Kiali CR to set `cluster_wide_access=true`.

**Fix:**
Every time the memberroll controller sets `accessible_namespaces="**"`, it should set explicitly also `cluster_wide_access=true`

### CODE REVIEW

File: pkg/controller/servicemesh/memberroll/controller.go [hunk 1]
```diff
@@ -711,7 +711,23 @@ func (r *defaultKialiReconciler) reconcileKiali(ctx context.Context, kialiCRName
 		return pkgerrors.Wrapf(err, "cannot set deployment.accessible_namespaces in Kiali CR %s/%s", kialiCRNamespace, kialiCRName)
 	}
 
-	err = updatedKiali.Spec.SetStringSlice("api.namespaces.exclude", excludedNamespacesSet.List())
+	// if the accessible namespaces include "**", set the cluster wide access flag to true
+	// since other combination (like accessible_namespaces="**" and cluster_wide_access=false) are not allowed in Kiali operator and reconsilation fails
```
Comment (jewertow) [inline pkg/controller/servicemesh/memberroll/controller.go:715]: This comment makes sense, but I am slightly confused, because the PR description says the opposite:
```
Every time the memberroll controller sets accessible_namespaces="**", it should set explicitly also cluster_wide_access=false
```
Could you please update the PR description?

File: pkg/controller/servicemesh/memberroll/controller_test.go [hunk 2]
```diff
@@ -810,6 +813,86 @@ func TestKialiResource(t *testing.T) {
...
+		{
+			name:                      "cluster-wide-access-false-when-regular-namespaces",
+			accessibleNamespaces:      []string{"bookinfo", "foo"},
+			expectedClusterWideAccess: false,
+		},
+		{
+			name:                      "cluster-wide-access-true-when-double-asterisk-with-other-namespaces",
+			accessibleNamespaces:      []string{"**", "bookinfo"},
+			expectedClusterWideAccess: true,
+		},
```
Comment (jewertow) [inline pkg/controller/servicemesh/memberroll/controller_test.go:836]: Could we add a test case for the empty accessible namespace set?

File: pkg/controller/servicemesh/memberroll/controller.go (hunk 1 above)
Comment (mkralik3) [reply pkg/controller/servicemesh/memberroll/controller.go:715]: yes, the PR description was incorrect. Fixed

**Comment by mkralik3** [comment]
@jewertow thanks for the review. Pushed changes

**Review by jewertow — APPROVED** [review]
LGTM
